docs/2.9.0/docs/query/pqs-user-guide.rst

```diff
@@ -505,988 +1780,2081 @@ Use a ``--config`` file to define multiple options or reflect an infrastructure-
+The health of the PQS process can be monitored using the health check
+endpoint ``/livez``. The health check endpoint is available at the
+``--health-port`` specified when launching PQS, which defaults to 8080.
 
-1. It should be acceptable to the user to sort results on this column.
-2. You need a (unique) B-tree index on this column.
-3. The column must have unique values.
+.. code:: bash
 
-You can then perform queries like this:
+   $ curl http://localhost:8080/livez
+   {"status":"ok"}
 
-.. code-block:: sql
+Redaction
+=========
 
-    SELECT *
-    FROM the_table
-    WHERE the_sort_col > ???
-    ORDER BY the_sort_col
-    LIMIT 100;
+The redaction feature enables removal of sensitive or personally
+identifiable information from contracts and exercises within the PQS
+database. This operation is particularly useful for complying with
+privacy regulations and data protection laws, as it enables the
+permanent removal of contract payloads, contract keys, choice arguments,
+and choice results. Note that redaction is a destructive operation and
+once redacted, information cannot be restored.
 
-The ``???`` value represents the last (largest) value for ``the_sort_col`` that was previously returned. To fetch results for the very first page, omit the ``WHERE`` clause.
+The redaction process involves assigning a ``redaction_id`` to a
+contract or an exercise and nullifying its sensitive data fields. For
+contracts, the ``payload`` and ``contract_key`` fields are redacted,
+while for exercises, the ``argument`` and ``result`` fields are
+redacted.
 
-Here is an example of random access to display page 10 of the search results:
+Conditions for Redaction
+------------------------
 
-.. code-block:: sql
+Redaction is allowed for:
 
-    SELECT *
-    FROM the_table
-    ORDER BY the_sort_col
-    LIMIT 100
-    OFFSET 1000;
+-  Contracts that are inactive (archived). ie. active contracts cannot
+   be redacted.
+-  Contracts that have not already been redacted.
+-  Any choice exercise
 
-This only makes sense if there is a B-tree index on ``the_sort_col``.
+A redaction operation requires a redaction ID, which is an arbitrary
+label to identify the redaction and provide information about its
+reason, and correlate with other systems that coordinate such activity.
 
-You should assume that a large ``OFFSET`` is slow. See the chapter on `LIMIT and OFFSET <https://www.postgresql.org/docs/current/queries-limit.html>`__ in the PostgreSQL manual.
+Examples
+--------
 
-psql tips
-=========
+Redacting an Archived Contract
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Type ``psql <dbname>`` on the command line to enter the PostgreSQL ```REPL``` (if in doubt, use postgres as the database name). Some useful commands are shown in the following table.
+To redact an archived contract, use the ``redact_contract`` function by
+providing the ``contract_id`` and a ``redaction_id``. This operation
+nullifies the ``payload`` and ``contract_key`` of the contract and
```

Comment:
I think we need to explain what 'nullifies' means because this is open to interpretation.